vllm/entrypoints/llm.py

```diff
@@ -251,4 +256,23 @@ def _run_engine(self, use_tqdm: bool) -> List[RequestOutput]:
         # This is necessary because some requests may be finished earlier than
         # its previous requests.
         outputs = sorted(outputs, key=lambda x: int(x.request_id))
-        return outputs
\ No newline at end of file
+        return outputs
+
+    def _validate_prompt(
+        self,
+        prompt: Optional[str] = None,
+        prompt_token_ids: Optional[List[int]] = None,
+        truncate_prompt_tokens: Optional[Annotated[int, Field(ge=1)]] = None
+    ) -> Tuple[Optional[str], Optional[List[int]]]:
+
+        if not (prompt or prompt_token_ids):
+            raise ValueError("Either prompt or prompt_ids should be provided.")
+        if (prompt and prompt_token_ids):
+            raise ValueError(
+                "Only one of prompt or prompt_ids should be provided.")
+        if truncate_prompt_tokens is not None:
+            if prompt_token_ids is None:
+                prompt_token_ids = self.llm_engine.tokenizer.tokenizer(
+                    prompt).input_ids
+            prompt_token_ids = prompt_token_ids[-truncate_prompt_tokens:]
```

Comment:
This isn't exactly the same behaviour as the [equivalent code ](https://github.com/vllm-project/vllm/blob/0650e5935b0f6af35fb2acf71769982c47b804d7/vllm/entrypoints/openai/serving_engine.py#L201-L210) in the completions API. In that case, we use the build-in truncation functionality inside the tokenizer in the case that the user provides a prompt, and only implement it ourselves in the case that the user provides token ids. I think it would be good to make it consistent here too.